tests/functional/tokens/test_login.py

```diff
@@ -0,0 +1,168 @@
+from pytest_bdd import given, scenario, then, when
+from pytest_bdd.parsers import parse
+
+
+@scenario("../../features/tokens/login.feature", "Login using RSTUF API")
+def test_login_using_rstuf_api():
+    """Login using RSTUF API."""
+
+
+@given(
+    parse(
+        "the API requester prepares 'data' with {username}, {password},"
+        "{scope}, {expires}"
+    ),
+    target_fixture="data",
+)
+def the_api_requester_prepares_data(username, password, scope, expires):
+
+    if expires == "None":
+        expires = None
+    else:
+        expires = int(expires)
+
+    return {
+        "username": username,
+        "password": password,
+        "scope": scope,
+        "expires": expires,
+    }
+
+
+@when(
+    "the API requester sends a 'POST' method to '/api/v1/token' with 'data'",
+    target_fixture="response",
+)
+def the_api_requester_sends_data(data, http_request):
+    return http_request(method="POST", url="/api/v1/token", data=data)
+
+
+@then("the API requester should get status code '200'")
+def the_api_requester_should_get_status_code_200(response):
+    assert response.status_code == 200
+
+
+@then("the API requester should get 'access_token' in response body")
+def the_api_requester_should_get_access_token_in_response_body(response):
+    assert response.json()["access_token"] is not None
+
+
+@scenario(
+    "../../features/tokens/login.feature",
+    "Login using RSTUF API gets Unauthorized",
+)
+def test_login_using_rstuf_api_gets_unauthorized():
+    """Login using RSTUF API gets Unauthorized."""
+
+
+@when(
+    "the API requester sends a 'POST' method to '/api/v1/token' with invalid "
+    "username/password in 'data'",
+    target_fixture="unauthorized_response",
+)
+def the_api_requester_sends_data_unauthorized(data, http_request):
+    return http_request(method="POST", url="/api/v1/token", data=data)
+
+
+@then("the API requester should get status code '401'")
+def the_api_requester_should_get_status_code_401(unauthorized_response):
+    assert unauthorized_response.status_code == 401
+
+
+@then(
+    "the API requester should get 'detail: Unauthorized' in the response body"
+)
+def the_api_requester_should_get_detail_unauthorized_in_the_response_body(
+    unauthorized_response,
+):
+    assert unauthorized_response.json()["detail"] == "Unauthorized"
+
+
+@scenario(
```

Comment:
While reviewing the code it was a little hard for me to make a separation between the different scenarios given how many small functions exists in the test file.

I suggest we separate the different scenarios in some way for example like this:
``` suggestion
    assert unauthorized_response.json()["detail"] == "Unauthorized"


################################## Scenario 2 ##################################


@scenario(
```

Of course, we can use a single new line between before and after the comment, but the idea is to have some visual separation.
Any thoughts?

Reply:
I agree it's a bit confusing, but if you follow the decorator (above comment, I mention the same), you will get used to it. Also, I recommend always having the .feature open to guide your reading.

Reply:
I have opened the `.feature` file opened, but still, it's confusing.
I feel we could still improve it with simple comments and separation.

Reply:
I added some separators. To be honest, I don't see much clarity with that.
In my case, I get highlighted by the `@scenario`
![Screenshot 2022-11-22 at 09 15 31](https://user-images.githubusercontent.com/546882/203261842-5db9df79-8c20-4d28-b9f1-7224743d91d6.png)

Reply:
As discussed, I'm reverting it.